v-next/hardhat-node-test-reporter/src/reporter.ts

```diff
@@ -138,6 +157,22 @@ export function hardhatTestReporter(
             break;
           }
 
+          // We do not want to display failures caused by the parent already
+          // having finished when running with the --test-only flag because
+          // they are false negatives.
+          // We still display the failed suites, because failures for the nested
+          // tests (cancelled by parent errors) were already displayed.
```

Comment:
This doesn't work, it looks like an error. I think a good trade-off would be to detect that situation and maybe display those canceled tests/suites as a particular case in the summary at the end of the test run if needed.

This is a complex case, so we can continue addressing it in a follow up PR.

Reply:
I replied in a longer form in https://github.com/NomicFoundation/hardhat/pull/5724#issuecomment-2343028639 TLDR I removed any attempt to handle that case from this PR + the newest node makes the output for these cases a lot better (e.g. in `hardhat` package, `test:only` works as expected now).